src/app/calendar/_components/CalendarTimeManager.tsx

```diff
@@ -0,0 +1,56 @@
+"use client";
+import { Dropdown } from "@/components/Dropdown";
+import { OrangeButton } from "@/components/OrangeButton";
+import { XButton } from "@/components/XButton";
+import { Dispatch, SetStateAction, useState } from "react";
+import { twMerge } from "tailwind-merge";
+
+type CalendarTimeManagerProps = {
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  className?: string;
+};
+
+export function CalendarTimeManager({
+  isOpen,
+  setIsOpen,
+  className,
+}: CalendarTimeManagerProps) {
+  const [selectedRoom, setSelectedRoom] = useState("1.38");
+  const availableRooms = ["2.41", "1.38", "3.33", "4.22"];
+  return (
+    <div
+      className={twMerge(
+        "z-0 sm:rounded-r-lg maxSM:rounded-b-lg bg-secondary relative",
+        !isOpen &&
+          "sm:w-0 maxSM:h-0 duration-700 transition-[width] maxSM:transition-[height]",
+        isOpen &&
+          "sm:w-full maxSM:h-full sm:-ml-2 maxSM:-mt-2 transition-[width] maxSM:transition-[height] duration-700"
+      )}
+    >
+      {isOpen && (
```

Comment:
You have duplicate `isOpen && ()` either merge it into one statement, or move it all to a seperate component with an early return (`if (!isOpen) return null`)